static/js/stream_edit.js

```diff
@@ -535,6 +536,7 @@ exports.initialize = function () {
     $("#subscriptions_table").on('click', '.close-privacy-modal', function (e) {
         // This fixes a weird bug in which, subscription_settings hides
         // unexpectedly by clicking the cancel button.
+        overlays.enable_background_mouse_events();
```

Comment:
Why isn't this `overlays.close_modal()` as well?

Reply:
Close icon and cancel button closes the modal by using data-dismiss=modal 
written in the template.
so when we use close_modal, the modal is already closed and the `is_modal_open` function returns false and further code of close_modal is not executed. 
One way will be to remove `data-dismiss=modal` and use close_modal.